From b816b72e3ccd53b684d072dc7e1247f661b0fb84 Mon Sep 17 00:00:00 2001
From: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Date: Thu, 13 Jun 2013 10:43:20 +0900
Subject: [PATCH 0482/1083] ARM: shmobile: r8a7790: Add RGX support

Add I/O resource data and platform_device information for RGX.

Signed-off-by: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
(cherry picked from commit 508b1190d6f9b82f31ee43af189af2c7b70de031)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7790.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/arch/arm/mach-shmobile/setup-r8a7790.c b/arch/arm/mach-shmobile/setup-r8a7790.c
index 21bbc51..6cea67e 100644
--- a/arch/arm/mach-shmobile/setup-r8a7790.c
+++ b/arch/arm/mach-shmobile/setup-r8a7790.c
@@ -119,6 +119,29 @@ static inline void r8a7790_register_scif(int idx)
 				      sizeof(struct plat_sci_port));
 }
 
+static struct resource powervr_resources[] = {
+	{
+		.start  = 0xfd000000,
+		.end    = 0xfd00ffff,
+		.flags  = IORESOURCE_MEM,
+	},
+	{
+		.start  = gic_spi(119),
+		.flags  = IORESOURCE_IRQ,
+	},
+};
+
+static struct platform_device powervr_device = {
+	.name           = "pvrsrvkm",
+	.id             = -1,
+	.resource       = powervr_resources,
+	.num_resources  = ARRAY_SIZE(powervr_resources),
+};
+
+static struct platform_device *r8a7790_early_devices[] __initdata = {
+	&powervr_device,
+};
+
 static struct renesas_irqc_config irqc0_data = {
 	.irq_base = irq_pin(0), /* IRQ0 -> IRQ3 */
 };
-- 
1.8.3.2

